### DISCUSSION TITLE
How to globally change logging format (to JSON)?

### DISCUSSION THREAD

**Discussion by mmutso-boku** [dagster-io/dagster#14258] [Q&A]
Hello,

What I'm trying to achieve: Have all dagster logs be JSON formatted using https://github.com/madzak/python-json-logger 

From going through the existing docs: https://docs.dagster.io/concepts/logging/loggers and https://docs.dagster.io/concepts/logging/python-logging I understand that this is not possible currently?

I tried out specifying the built-in json_console_logger in the Definitions, as described here https://docs.dagster.io/concepts/logging/loggers#specifying-default-code-location-loggers but I only managed to get it to work when running a job in launchpad and defining the config for the logger. Can't figure out, how could this be used when materializing an asset (aka not creating a separate job from the asset).

Looking at the Python logging section https://docs.dagster.io/concepts/logging/python-logging it also seems, that this does not cover what I am trying to do, since Dagster creates the handler and formatter itself, but I would like it to reuse the pythonjsonlogger.jsonlogger.JsonFormatter.

So, am I missing something and it is possible to globally make dagster logs be JSON (without adding some config to every single asset/job)? Or this is something that would be solved with https://github.com/dagster-io/dagster/discussions/12495?

Thanks in advance

**Comment by smackesey** [comment]
Hi @mmutso-boku, our current logging configuration is a little wonky and going to get overhauled. There is a broader discussion about it [here](https://github.com/dagster-io/dagster/discussions/12495).

cc @rexledesma for an answer to whether it is currently impossible to globally change logging to JSON.
